### PR TITLE
User search

### PR DESCRIPTION
**Pull request by ice-orion**

### CODE REVIEW

**Review by victor-oancea-leftclick-io — CHANGES_REQUESTED** [review]
Would have been a great PR if all features were implemented

File: src/components/UserListItem/components/UserListPingButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import {PingIcon} from '@svg/PingIcon';
+import {t} from '@translations/i18n';
+import React from 'react';
+
+type Props = {
+  pinged?: boolean | null;
+};
+
+export const UserListPingButton = ({pinged}: Props) => {
+  if (pinged == null) {
```
Comment (ice-ares) [inline src/components/UserListItem/components/UserListPingButton/index.tsx:14]: or undefined

File: src/utils/errors.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import {t} from '@translations/i18n';
+import {checkProp} from '@utils/guards';
+
+export const getErrorMessage = (error: unknown): string => {
+  if (checkProp(error, 'message')) {
+    return String(error.message);
+  } else if (typeof error === 'string') {
+    return error;
+  }
+  return t('error.general_error');
```
Comment (ice-ares) [inline src/utils/errors.ts:12]: or check if its API error, and in that case return some default message + code from resp body

File: src/components/UserListItem/components/UserListPingButton/index.tsx (hunk 1 above)
Comment (ice-orion) [reply src/components/UserListItem/components/UserListPingButton/index.tsx:14]: `==` is used intentionally to match `undefined`, so 
```
undefined == null -> true
null == null -> true
```

File: src/utils/errors.ts (hunk 2 above)
Comment (ice-orion) [reply src/utils/errors.ts:18]: done, in case of isApiError->true the message is `Request failed with status {{status}} and code {{code}}`

**Review by ice-ares — APPROVED** [review]

**Review by ice-eros — APPROVED** [review]

**Review by ice-hector — APPROVED** [review]

**Pull request merged**